tests/test_set_password.py

```diff
@@ -43,8 +42,8 @@ def test_token_does_not_match(self):
 
     def test_no_activation_pending(self):
         """Expect an error if an account is activated for the seconds time."""
-        db.session.add(User("SomeUser", "SomeToken",
-                            "SomeMeterId", "SomeGroup"))
+        db.session.add(User(GenderType.MALE, "SomeUser", "User@Some.net",
```

Comment:
Do you really want to create a user with the same parameters as in "test_user_does_not_exist()"? I'd suggest creating all user instances needed for the different test functions in a setUp() method to avoid duplicate code.